bucket/imagemagick.json

```diff
@@ -17,29 +17,14 @@
         }
     },
     "innosetup": true,
-    "bin": [
-        "compare.exe",
-        "composite.exe",
-        "conjure.exe",
-        [
-            "convert.exe",
-            "imconvert"
-        ],
-        "dcraw.exe",
-        "identify.exe",
-        "IMDisplay.exe",
-        "magick.exe",
-        "mogrify.exe",
-        "montage.exe",
-        "stream.exe"
-    ],
     "env_set": {
         "MAGICK_HOME": "$dir",
         "MAGICK_CODER_MODULE_PATH": "$dir\\modules\\coders"
     },
     "env_add_path": ".",
```

Comment:
Is there an advantage of adding to path over bin?

Is it possible for us to add DLLs to bin?

Is it possible for us to copy the DLLs to a totally new folder and add that to bin?

Reply:
I don't think e can create shims for DLLs. 

The number and names of DLLs is likely to be non-static, it might be problematic in the future if we hardcode them now.

Reply:
The issue with adding to path is that we end up adding every new dependency present in them as well.

For example, if they add a future binary which causes a convert like issue that can be problematic.

Also renaming convert is likely to cause issue for scripts which might use convert.exe from MAGICK_HOME directory.

Are we sure we are not breaking workflows?

Reply:
The internal Windows `convert` utility is usually present higher up in PATH (because that PATH segment comes from system-wide environment instead of user-wide environment). So any scripts that do use `convert`, will use the internal `convert` (unless of course they specify the full path). I think this is a fair assumption we can make.

Reply:
> internal convert (unless of course they specify the full path)

scripts which would use an environment variable like MAGICK_HOME are more than likely to use and specify the full path due to the creators being aware of the collision of names between the ImageMagick convert and Windows convert.

And if there is a low chance of collision as you mention, why rename it anyways?

Why not just add it to bin?

If existing Scoop based scripts using imconvert exists, we ensure they don't break and existing MAGICK_HOME based scripts too won't break.

Reply:
I think a more experienced ImageMagick user like @tech189 might be able suggest better here...

Reply:
> How about we leave a note for those who upgrade to raise issues on Github?

I don't see this as the best of ideas.

Reply:
I agree with @tech189 on the fact that most people will use ImageMagick through `magick <subcommand> ...` like `magick convert ...`, so maybe renaming it isn't that big of a deal (also the convert utility has been deprecated as tech189 said).

Reply:
Ok just to confirm, I've kept the shims just in case and left a note to users to use 'magic convert ...' instead. We can remove the shims at a later stage if someone encounters issues with the duplicates.

> I don't see this as the best of ideas.

Sure, instead I'll monitor the issues and @ the owner of the rinohtype repo mentioned above to check their workflows are okay.

Otherwise, I think we're all done now?

Reply:
Wait, the issue was only in the `convert.exe` shim, not others (as they were available both before this PR and will be available after the PR - they won't break any workflows)

Reply:
That's right, I'm not touching any of the other shims. I'm just removing the convert.exe shim and renaming the binary to imconvert.exe to avoid conflicts.

Reply:
Um no no, I think there's a misunderstanding. The problem of having duplicated binaries is still unsolved. 

Let me fix it.

Reply:
Done

Reply:
Ah okay, looks good to me. Just tested it. Sorry for the misunderstanding.

Reply:
No worries 😊